pubtools/_content_gateway/push_staged.py

```diff
@@ -1,2 +1,191 @@
+from pushsource import Source, CGWPushItem
```

Comment:
The CGWPushItem is the class that we have to check the instance against. And sources get a source of content.
Check for more details https://github.com/release-engineering/pushsource#readme

Reply:
h, I meant push_staged and push_cgw

Reply:
`push_cgw` will be called through CLI entry point by passing the yaml file like:- `push-cgw-metadata --CGW_hostname abc.com -p pwd -u username -f file_path/cgw_push.yml` so the cgw operation will be performed on the data received from the YAML file. 
On the other hand, the `push_staged` will be called from pub methods by setting up the targets. So it will operate on the data received from the pub server. The push_staged may need some extra methods in the future so I decided to keep both classes independent to avoid tight coupling.

Reply:
I don't know if this kind of duplication is good design. If you think that extra methods will be necessary for push_staged in the future, perhaps you could create a subclass?

Reply:
At present, I'm not sure what else will be added to the push_staged so I believe it's a good idea to keep it independent for now as both the classes need different parameters to initiate and down the line they may vary. If we will not have many changes further will try to merge them together.